### PR TITLE
Validating plotting configuration

### LINKED ISSUE #323: Validate topostats/plotting_dictionary.yaml
After having introduced validation of `topostats/default_config.yaml` we should also validate the plotting dictionary (`topostats/plotting_dictionary.yaml`) to ensure that if people modify it that it is valid.

### PR DESCRIPTION
**Pull request by ns-rse**
Closes #323 

Adds `io.validate_plotting()` function and incorporates it into `run_topostats.py` so that the plotting dictionary/YAML configuration is validated. I also took the opportunity to add a command line argument for users to provide their own plotting configuration YAML file.

Also noticed that whilst I had removed `output_dir` from `Grains()` in #325 as it was no longer used and should have been in `run_topostats.py` that on running through `flake8` courtesy of `pre-commit` it threw up the following error...

```
topostats/run_topostats.py:178:9: F841 local variable 'grain_out_path' is assigned to but never used
```

And so I've added a line to ensure that the `grain_out_path` is actually created (I should have had these checks in place much earlier, hope others are using them and finding them useful).

For some reason the `topostats/default_config.yaml` also had `channel` under `filters` which I thought I'd removed previously in #325 but despite `git checkout dev && git pull && git checkout -b ns-rse/323-validating-plotting` it reappeared. This meant it failed the validation. Hopefully it won't creep back in again!

### CODE REVIEW

File: topostats/validation.py [hunk 1]
```diff
@@ -97,23 +117,30 @@ def validate_config(config: dict):
...
+                    True,
+                    False,
+                    error="Invalid value in config for 'plotting.run', valid values are 'True' or 'False'",
                 ),
                 "save_format": str,
                 "image_set": Or(
                     "all",
                     "core",
-                    error="Invalid value in config for 'plotting.image_set', valid values are 'all' or 'core'",
+                    error="Invalid value in config for 'plotting.image_set', valid values " "are 'all' or 'core'",
```
Comment (SylviaWhittle) [inline topostats/validation.py:143]: Is this addition of closing and opening quotes " intended, if so, why did you add it? Just curious!

**Review by SylviaWhittle — APPROVED** [review]
I've read through the code changes and I've tried editing the plotting file in unsupported ways, causing the code to successfully tell me that there was an error in the configuration. Besides that I wasn't sure how to test this PR. If there is anything else you would like me to test with it, let me know 👍

File: topostats/validation.py (hunk 1 above)
Comment (ns-rse) [reply topostats/validation.py:143]: I did it in other places because the file was failing linting with `flake8` due to long lines, doesn't need it on this line though, good spot, I'll remove it in in a separate PR with some other work.
